src/fwriteR.c

```diff
@@ -5,18 +5,18 @@
 #define DATETIMEAS_EPOCH     2
 #define DATETIMEAS_WRITECSV  3
 
-#define NEED2NATIVE(s) !(IS_ASCII(s) || (s)==NA_STRING)
-#define ENCODED_CHAR(s) (utf8 && NEED2UTF8(s) ? translateCharUTF8(s) : \
-  (native && NEED2NATIVE(s)) ? translateChar(s) : CHAR(s))
+static bool utf8=false;
+static bool native=false;
+#define TO_UTF8(s) (utf8 && NEED2UTF8(s))
+#define TO_NATIVE(s) (native && (s)!=NA_STRING && !IS_ASCII(s))
+#define ENCODED_CHAR(s) (TO_UTF8(s) ? translateCharUTF8(s) : TO_NATIVE(s) ? translateChar(s) : CHAR(s))
```

Comment:
thanks, now is more readable, still it is not obvious if
```
TO_UTF8(s) ? translateCharUTF8(s) : TO_NATIVE(s)
```
is meant to evaluate to boolean condition for second ternary operator.
If we just add extra `(` there it is then very clear that it isn't:
```
 (TO_UTF8(s) ? translateCharUTF8(s) : (TO_NATIVE(s) ? translateChar(s) : CHAR(s)))
```

Reply:
Ha. I sometimes write this kind of multiple ternary operator, splitting to multiple lines actually looks clear to me (maybe I just get used to it). I'm ok to add the extra `()`.

Reply:
Thank you for your patience, I can imagine so many minor stuff can be annoying :)